sisyphus/task.py

```diff
@@ -3,35 +3,35 @@
 import logging
 import sys
 import time
-import typing
+from typing import Optional, Union, Any, Sequence, Dict, List
 import subprocess as sp
 from ast import literal_eval
 
 import sisyphus.tools as tools
 import sisyphus.global_settings as gs
 
 
-class Task(object):
+class Task:
     """
     Object to hold information what function should be run with which requirements.
     """
 
     def __init__(
         self,
-        start,
-        resume=None,
-        rqmt=None,
-        args=None,
-        mini_task=False,
+        start: str,
+        resume: Optional[str] = None,
+        rqmt: Optional[Dict[str, Any]] = None,
+        args: Optional[Sequence[Union[List[Any], Any]]] = None,
```

Comment:
```suggestion
        args: Optional[Sequence[Any]] = None,
```
can already be `Any` and `List[Any]` is contained in `Any`.

Reply:
Yea I know, but sometimes I prefer this to give some more specific type hints to what types would be specifically expected here, or specially handled, even though other types (any other types) would also be ok.

Also, I basically just translated the typing from the earlier docstring. I also wondered a bit about why this is here at all, and I thought there is indeed maybe a special meaning when there is a list?